nemo/collections/nlp/parts/mixins/nlp_adapter_mixins.py

```diff
@@ -88,8 +88,10 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
         self.use_mcore_gpt = hasattr(self, 'mcore_gpt') and self.mcore_gpt
-        if self.use_mcore_gpt:
-            assert HAVE_MEGATRON_CORE, "You set `mcore_gpt` as True but megatron core is not found."
+        self.use_mcore_t5 = hasattr(self, 'mcore_t5') and self.mcore_t5
+
+        if self.use_mcore_gpt or self.use_mcore_t5:
+            assert HAVE_MEGATRON_CORE, "You set `mcore_gpt` or `mcore_t5` as True but megatron core is not found."
 
     def _unwrap_model(self):
         if not hasattr(self, "model"):
```

Comment:
do you not need to change this for T5?

Reply:
We don't need to set `enc_dec_model` here, because in `MegatronT5SFTModel` class, there is already a property (managed attribute) `model` added to the class. 
(https://github.com/NVIDIA/NeMo/blob/08cc5158274bef7c7b2bb8ae93be184ad111ed80/nemo/collections/nlp/models/language_modeling/megatron_t5_sft_model.py#L157)
So I just keep the `self.model` usage to keep it consistent with the rest of the code.